### PR TITLE
Oracle IP/CS providers: Set OCI SDK to retrieve region info from IMDS

### PR DESCRIPTION
**Pull request by harryjsmith**
This ensures that we're using the correct second-level domain for the auth endpoint in these environments

### CODE REVIEW

File: provider/oracle/cloudshell_provider.go [hunk 1]
```diff
@@ -87,6 +87,9 @@ func NewCSProvider(configSource provider.ConfigSource, passphraseSource provider
 		compartmentID = csConfig.tenancyID
 	}
 
+	// Set OCI SDK to use IMDS to fetch region info (second-level domain etc.)
+	oci.EnableInstanceMetadataServiceLookup()
```
Comment (templecloud) [inline provider/oracle/cloudshell_provider.go:91]: Cool. Nice one.

File: provider/oracle/ip_provider.go [hunk 2]
```diff
@@ -5,13 +5,16 @@ import (
...
-	"github.com/oracle/oci-go-sdk/common/auth"
+	"github.com/oracle/oci-go-sdk/v27/common"
+	"github.com/oracle/oci-go-sdk/v27/common/auth"
 
 	"github.com/fnproject/fn_go/provider"
 )
 
 func NewIPProvider(configSource provider.ConfigSource, passphraseSource provider.PassPhraseSource) (provider.Provider, error) {
+	// Set OCI SDK to use IMDS to fetch region info (second-level domain etc.)
+	common.EnableInstanceMetadataServiceLookup()
```
Comment (templecloud) [inline provider/oracle/ip_provider.go:16]: Cool. Nice one.

**Review by templecloud — COMMENTED** [review]
Looks good to me. So we have a new version of the oci SDK and enabling a property on that to use the second level domain form the metadata service of the environment it is in?

**Review by templecloud — APPROVED** [review]
LGTM if the test results come in good.

**Review by riconnon — APPROVED** [review]

**Comment by harryjsmith** [comment]
> Looks good to me. So we have a new version of the oci SDK and enabling a property on that to use the second level domain form the metadata service of the environment it is in?

Correct

**Pull request merged**